smart_open/tests/test_s3.py

```diff
@@ -34,29 +34,30 @@ def maybe_mock_s3(func):
         return moto.mock_s3(func)
 
 
-def cleanup_bucket(s3, delete_bucket=False):
-    for bucket in s3.buckets.all():
-        if bucket.name == BUCKET_NAME:
-            for key in bucket.objects.all():
-                key.delete()
-
-            if delete_bucket:
-                bucket.delete()
-                return False
-            return True
-    return False
-
-
-def create_bucket_and_key(
-        bucket_name=BUCKET_NAME, key_name=KEY_NAME, contents=None,
-        num_attempts=12, sleep_time=5):
-    # fake (or not) connection, bucket and key
-    logger.debug('%r', locals())
+@maybe_mock_s3
+def setUpModule():
+    global s3
```

Comment:
Is it possible to avoid using a global? It's a bit of a code smell.

If it's not possible, then please add a comment about these two functions. Things to mention:

- They're special functions that the unittest library looks for
- setUpModule gets called before any code from the module is run (?)
- tearDownModule gets called ... (when?)
- The motivation (create a new bucket when the module is loaded, and delete the bucket when ...

It's also worth making `s3` more obviously a global. Something like `GLOBAL_S3` would be better. That way, when people touch it, they'll at least know what they're doing.